test/index.test.ts

```diff
@@ -183,15 +236,450 @@ describe('HipThrusTS', () => {
+          typeof pipedLeftOnly.attachData
+        >[0]
+          ? true
+          : false;
+        type ParamAssignableFromCorrect = Parameters<
+          typeof pipedLeftOnly.attachData
+        >[0] extends CorrectParam
+          ? true
+          : false;
+        type ReturnAssignableToCorrect = CorrectReturnValue extends ReturnType<
+          typeof pipedLeftOnly.attachData
+        >
+          ? true
+          : false;
+        type ReturnAssignableFromCorrect = ReturnType<
+          typeof pipedLeftOnly.attachData
+        > extends CorrectReturnValue
+          ? true
+          : false;
+        // @ts-expect-error
+        const paramAssignableToCorrectShouldFail: ParamAssignableToCorrect = false;
+        // @ts-expect-error
+        const paramAssignableFromCorrectShouldFail: ParamAssignableFromCorrect = false;
+        // @ts-expect-error
+        const returnAssignableToCorrectShouldFail: ReturnAssignableToCorrect = false;
+        // @ts-expect-error
+        const returnAssignableFromCorrectShouldFail: ReturnAssignableFromCorrect = false;
+      });
+      it('attaches properly typed data from right sync data attacher only', () => {
+        const rightPartiallyCovered = {
+          attachData(context: { b: number; other: string }) {
+            return { c: 4 };
+          },
+        };
+
+        const pipedRightOnly = HTPipeAttachData({}, rightPartiallyCovered);
+
+        interface CorrectParam {
+          b: number;
+          other: string;
+        }
+        type CorrectReturnValue = PromiseOrSync<{ c: number }>;
+        type ParamAssignableToCorrect = CorrectParam extends Parameters<
+          typeof pipedRightOnly.attachData
+        >[0]
+          ? true
+          : false;
+        type ParamAssignableFromCorrect = Parameters<
+          typeof pipedRightOnly.attachData
+        >[0] extends CorrectParam
+          ? true
+          : false;
+        type ReturnAssignableToCorrect = CorrectReturnValue extends ReturnType<
+          typeof pipedRightOnly.attachData
+        >
+          ? true
+          : false;
+        type ReturnAssignableFromCorrect = ReturnType<
+          typeof pipedRightOnly.attachData
+        > extends CorrectReturnValue
+          ? true
+          : false;
+        // @ts-expect-error
+        const paramAssignableToCorrectShouldFail: ParamAssignableToCorrect = false;
+        // @ts-expect-error
+        const paramAssignableFromCorrectShouldFail: ParamAssignableFromCorrect = false;
+        // @ts-expect-error
+        const returnAssignableToCorrectShouldFail: ReturnAssignableToCorrect = false;
+        // @ts-expect-error
+        const returnAssignableFromCorrectShouldFail: ReturnAssignableFromCorrect = false;
+      });
+      it('no attaches data when left and right is empty objects', () => {
+        const pipedRightOnly = HTPipeAttachData({}, {});
+
+        // tslint:disable-next-line:prefer-const
+        let returnTypeOfHTPipeAttachData: {};
+
```

Comment:
where's the rest of this test?  It should have the 4 assertions too.
- [ ] where are the 4 assertions?

Reply:
- [ ] fix these names too
pipedRightOnly?